homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,282 @@
+            entry.data[CONF_MAC_CODE],
+            entry.data[CONF_BLIND_TYPE],
+            BLIND_TO_ENTITY_TYPE[entry.data[CONF_BLIND_TYPE]].__name__,
+        )
+        super().__init__()
+        self._device = device
+        self._device.register_running_callback(self.async_update_running)
+        self._device.register_position_callback(self.async_update_position)
+        self._device.register_connection_callback(self.async_update_connection)
+        self.entity_description = COVER_TYPES[entry.data[CONF_BLIND_TYPE]]
+        self.config_entry: ConfigEntry = entry
+
+        self._attr_name: str = device.display_name
+        self._attr_unique_id: str = entry.data[CONF_ADDRESS]
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            connections={(CONNECTION_BLUETOOTH, entry.data[CONF_ADDRESS])},
+            manufacturer=MANUFACTURER,
+            model=entry.data[CONF_BLIND_TYPE],
+            name=self._attr_name,
+        )
+
+    async def async_update(self) -> None:
+        """Update state, called by HA if there is a poll interval and by the service homeassistant.update_entity."""
+        _LOGGER.debug("(%s) Updating entity", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.connect()
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop moving the blind."""
+        _LOGGER.debug("(%s) Stopping", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.stop()
+
+    @callback
+    def async_update_running(
+        self, running_type: MotionRunningType | None, write_state: bool = True
+    ) -> None:
+        """Update whether the blind is running (opening/closing) or not."""
+        self._attr_is_opening = (
+            False
+            if running_type
+            in [None, MotionRunningType.STILL, MotionRunningType.UNKNOWN]
+            else running_type is MotionRunningType.OPENING
+        )
+        self._attr_is_closing = (
+            False
+            if running_type
+            in [None, MotionRunningType.STILL, MotionRunningType.UNKNOWN]
+            else running_type is not MotionRunningType.OPENING
+        )
+        if running_type is not MotionRunningType.STILL:
+            self._attr_is_closed = None
+        if write_state:
+            self.async_write_ha_state()
+
+    @callback
+    def async_update_position(
+        self,
+        position: int,
+        tilt: int,
+    ) -> None:
+        """Update the position of the motor."""
+        self._attr_current_cover_position = 100 - position
+        self._attr_current_cover_tilt_position = 100 - round(100 * tilt / 180)
+        self._attr_is_closed = self._attr_current_cover_position == 0
+        self.async_write_ha_state()
+
+    @callback
+    def async_update_connection(self, connection_type: MotionConnectionType) -> None:
+        """Update the connection status."""
+        if connection_type is MotionConnectionType.DISCONNECTED:
```

Comment:
would it make sensor for the entity to be unavailable in that case?

Reply:
Then you would not be able to control the cover right? Currently the blind is connected to once you move the cover entity. In the next PR I will include a button which you can also press to manually connect to the blind without moving the cover entity

Reply:
correct. But if it's disconnected it should not be controllable, so I think being unaviable makes sense, don't you?

Reply:
But then how would you connect to the blind since there is currently no button entity available?

Reply:
button entity is a separate entity, that can be available

Reply:
Sure but that is a second platform and the initial PR can only contain one platform right? Either way, I think it is a good feature for the blind to automatically connect once you move the cover, only to move after a couple of seconds when the blind is connected to. Otherwise you would have to press the button each time you want to connect, which is a good feature to have for when you simply want to update it's state for example but not control it

Reply:
Ideally the integration should always be connected to the device (e.g. by sending status requests in the background every x seconds). If this is not possible or desired for some reason, that would mean the device should be treated as always being disconnected with the state always being assumed. In that case I would not set the entity to unavailable as that would be the case pretty much all the time. Is there no way to keep a stable connection to the device? If the reason for the disconnect can be determined as being e.g. an empty battery, faulty adapter or misconfiguration of some kind, then the entity should be set to unvailable.

Reply:
That would unfortunately severely drain the battery of the motor and is not recommended. Users can choose to update the states themselves by calling `homeassistant.update_entity` which will connect to the blind and ask for the status. For wired motors I could include a permanent connection in a future pull request

Reply:
I'll probaby just remove this and handle it from the library by calling the callback `async_update_position(None, None)` if the device has disconnected, then the cover entity doesn't need to register a callback for connection updates
```
    @callback
    def async_update_connection(self, connection_type: MotionConnectionType) -> None:
        """Update the connection status."""
        if connection_type is MotionConnectionType.DISCONNECTED:
```

Reply:
I don't think it's a good idea to completely rely on the user having to update the entity for themselves (not that you are doing exactly this right now, I didn't really look at the relevant parts of the code). As long as there is some kind of automatic state update that doesn't have to be initiated manually by calling a service or pressing a button, but happens when the user is trying to alter the device via Home Assistant it's fine I guess. The only problem then would be if the device's state changes outside of Home Assistant, then those changes wouldn't be reflected until some command is executed. This is not ideal and perhaps it would be a good idea to make this behaviour configurable in a future PR e.g. by allowing the user to set a scan interval for automatic status updates.

Reply:
A future PR could indeed possibly add the option of setting a scan interval or enabling a permanent connection (for example for wired motors where battery life is not an issue), but this integration's goal is mainly to be used like the Motionblinds BLE mobile app which means the blind is only connected to when it is controlled